great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -400,6 +404,13 @@ def get_domain_records(
             else:
                 raise ValidationError(f"Unable to find batch with batch_id {batch_id}")
 
+        filter_conditions: List[RowCondition] = domain_kwargs.get(
+            "filter_conditions", []
```

Comment:
@anthonyburdi Is `filter_condition` a new key, which replaces some older keys?

Reply:
It's a new key that doesn't replace any existing keys that I am aware of or able to find. My hope is that we can eventually do away with this method and eventually specify RowConditions directly but this PR is meant as a first pass at enabling the functionality.